File: android/src/newarch/com/kk/rnumsdk/RNUmsdk.java
package com.kk.rnumsdk;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReadableMap;

import javax.annotation.Nullable;

public class RNUmsdk extends NativeUMSdkModuleSpec{

    public RNUmsdkImpl sdkImpl = null;
    public static ReactApplicationContext reactContext;
    public RNUmsdk(ReactApplicationContext context) {

        super(context);
        reactContext = context;
        sdkImpl = new RNUmsdkImpl(context);
    }

    @Override
    public String getName() {
        return RNUmsdkImpl.NAME;
    }

    @Override
    public void preInitUMSDK(String appkey, String channel){
        sdkImpl.preInitUMSDK(appkey, channel);
    };

    @Override
    public void initUMSDK(String appkey, String channel, String secret){
        sdkImpl.initUMSDK(appkey, channel, secret);
    };
    @Override
    public void getDeviceToken(Callback callback){
        sdkImpl.getDeviceToken(callback);
    };
    @Override
    public void getNonification(Callback callback){
        sdkImpl.getNonification(callback);
    };
    @Override
    public void auth(String platformType, Callback callback){
        sdkImpl.auth(reactContext.getCurrentActivity(), platformType, callback);
    };
    @Override
    public void shareToPlatform(String platformType, String shareType, ReadableMap params, @Nullable Callback callback){
        sdkImpl.shareToPlatform(reactContext.getCurrentActivity(), platformType, shareType, params, callback);
    };
    @Override
    public void isInstall(String platformType, Callback callback){
        sdkImpl.isInstall(reactContext.getCurrentActivity(), platformType, callback);
    };
    @Override
    public void onPageStart(String pageName) {
        sdkImpl.onPageStart(pageName);
    }
    @Override
    public void onPageEnd(String pageName) {
        sdkImpl.onPageEnd(pageName);
    }
    @Override
    public void profileSignInWithPUID(String puid) {
        sdkImpl.profileSignInWithPUID(puid);
    }
    @Override
    public void profileSignOff() {
        sdkImpl.profileSignOff();
    }
}

File: ios/brige/RNUMSdk.m

#import "RNUMSdk.h"
#import "RNUMSdkImpl.h"
#import "RNUMShare.h"
#import <UMCommon/UMCommon.h>
@implementation RNUMSdk

#if RCT_NEW_ARCH_ENABLED

#else
RCT_EXPORT_MODULE(RNUMSdkBridge)

RCT_REMAP_METHOD(preInitUMSDK, preInitUMSDK:(NSString *)appkey channel:(NSString *)channel){

}

RCT_REMAP_METHOD(initUMSDK, initUMSDK:(NSString *)appkey channel:(NSString *)channel secret:(NSString *)secret){
    [[RNUMSdkImpl sharedInstanceDelegate] initUMSDK:appkey channel:channel secret:secret];
}

RCT_REMAP_METHOD(getDeviceToken, getDeviceToken:(RCTResponseSenderBlock)callback){
    [[RNUMSdkImpl sharedInstanceDelegate] getDeviceToken:callback];
    
}
RCT_REMAP_METHOD(getNonification, getNonification:(RCTResponseSenderBlock)callback){
    [[RNUMSdkImpl sharedInstanceDelegate] getNonification:callback];
}
RCT_REMAP_METHOD(shareToPlatform, shareToPlatform:(NSString *)platformType shareType:(NSString *)shareType params:(NSDictionary *)params){
    [[RNUMSdkImpl sharedInstanceDelegate] shareToPlatform:platformType shareType:shareType params:params];
}
RCT_REMAP_METHOD(isInstall, isInstall:(NSString *)platformType callback:(RCTResponseSenderBlock)callback){
    [[RNUMSdkImpl sharedInstanceDelegate] isInstall:platformType callback:callback];

}
RCT_EXPORT_METHOD(auth:(NSString *)platformType callback:(RCTResponseSenderBlock)callback){
    [[RNUMSdkImpl sharedInstanceDelegate] auth:platformType callback:callback];
}
RCT_REMAP_METHOD(onPageStart, onPageStart:(NSString *)pageName){
    [[RNUMSdkImpl sharedInstanceDelegate] onPageStart:pageName];
}
RCT_REMAP_METHOD(onPageEnd, onPageEnd:(NSString *)pageName){
    [[RNUMSdkImpl sharedInstanceDelegate] onPageEnd:pageName];
}
RCT_REMAP_METHOD(profileSignInWithPUID, profileSignInWithPUID:(NSString *)puid){
    [[RNUMSdkImpl sharedInstanceDelegate] profileSignInWithPUID:puid];
}
RCT_REMAP_METHOD(profileSignOff, profileSignOff){
    [[RNUMSdkImpl sharedInstanceDelegate] profileSignOff];
}

#endif
- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}

//- 友盟
+ (void)initWithAppkey:(NSString* )umAppKey{
    [UMConfigure initWithAppkey:umAppKey channel:@"App Store"];
  
}

@end
  
